packages/containers/src/cmfConnect.js

```diff
@@ -0,0 +1,165 @@
+import React, { PropTypes, createElement } from 'react';
+import hoistStatics from 'hoist-non-react-statics';
+import { api } from 'react-cmf';
+import { connect } from 'react-redux';
+
+import {
+	statePropTypes,
+	initState,
+	getStateAccessors,
+	getStateProps,
+} from './state';
+
+export function getComponentName(WrappedComponent) {
+	return WrappedComponent.displayName || WrappedComponent.name || 'Component';
+}
+
+export function getComponentId(componentId, props) {
+	if (typeof componentId === 'function') {
+		return componentId(props) || 'default';
+	} else if (typeof componentId === 'string') {
+		return componentId;
+	} else if (props.componentId) {
+		return props.componentId;
+	}
+	return 'default';
+}
+
+export function getStateToProps({
+	componentId,
+	ownProps,
+	state,
+	mapStateToProps,
+	WrappedComponent,
+}) {
+	let props = getStateProps(
+		state,
+		getComponentName(WrappedComponent),
+		getComponentId(componentId, ownProps),
+	);
+	if (mapStateToProps) {
+		props = Object.assign(props, mapStateToProps(state, ownProps));
+	}
+	return props;
+}
+
+export function getDispatchToProps({
+	defaultState,
+	dispatch,
+	componentId,
+	mapDispatchToProps,
+	ownProps,
+	WrappedComponent,
+}) {
+	let props = getStateAccessors(
+		dispatch,
+		getComponentName(WrappedComponent),
+		getComponentId(componentId, ownProps),
+		defaultState,
+	);
+	props.dispatch = dispatch;
+	if (mapDispatchToProps) {
+		props = Object.assign(props, mapDispatchToProps(dispatch, ownProps));
```

Comment:
(see comments on `getStateToProps`) Same here.
```
// here cmfProps is your props variable

const userProps = mapDispatchToProps(dispatch, ownProps, cmfProps);
cmfProps = { ...cmfProps, ...userProps };
```

This would be an entry point to customise CMF state accessors functions (not encouraged but possible).
Ex : adding a log before calling the cmf function.